ref/general/microshed.adoc

```diff
@@ -6,78 +6,87 @@
+
+Integration testing evaluates the performance of a group of microservices as a whole unit, as compared to unit tests, which tests each microservice as an individual unit.
+Thus, integration tests are more comprehensive than unit tests.
+However, integration tests can be more complicated and take longer to run.
+Microshed testing makes your work easier and faster by streamlining integration tests with testcontainers.
 
-Integration testing looks at microservices as a group and tests them as a whole unit.
-Integration testing offers advantages over other types of testing, for example, detecting bugs earlier in the cycle.
 
 == Why integration testing
 
-Integration testing follows unit testing.
-After individual modules are unit tested, integration testing looks at the modules together to evaluate their performance as a group.
+You need to be sure that your application works in development as well as production mode.
+Developers usually run unit tests as they are easier, faster, and test a few methods at a time.
+Integration tests assess multiple test classes and components at once, and are better at analyzing the working of the different parts of your application as a whole.
+Therefore, developers find integration testing to be more reliable for their application to work in production environment.
+However, integration tests take longer and can be complicated to set up and configure.
+With faster development cycles and limited time available to developers, it's not always feasible to run integrated tests for your applications.
 
-== What challenges does MicroShed resolve
+== Development-production parity
 
-An application might run differently in production as compared to the development and testing environment.
+Development-production parity, or dev-prod parity, is one of the factors in the link:https://12factor.net/[twelve-factor app], a methodology to build modern applications.
+The idea behind dev-prod parity is to keep the development, staging, and production environments as similar as possible, and it addresses gaps in development ad production in terms of time, personnel, and tools.
+Developers mostly use different tools in development as compared to production for speed and ease of use.
+For example you might use a local maven build in development, but the application might be deployed to a docker container in production.
 Differences between the environments can cause a test to fail in production, even though it passed in development.
-You can address this issue by testing your application in the same container that you use in production.
-
-Running tests in development mode is convenient for local development, but it can be tedious to test against a running Open Liberty server in production scenarios, such as CI/CD pipelines.
-For this reason, MicroShed Testing can start and stop the application runtime before and after the tests are run.
-
-== How it works
+The application might work on your machine but not in production.
```

Comment:
However, the only way to ensure that what works in your local environment also works after deployment is to use the same services in development as you do in production.